fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/KeyStoreSerializationKeyManager.java

```diff
@@ -0,0 +1,231 @@
+        return cipherName;
+    }
+
+    @Override
+    public Random getRandom(int keyNumber) {
+        return secureRandom;
+    }
+
+    @Nonnull
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        @Nullable
+        String keyStoreFileName;
+        @Nullable
+        String keyStorePassword;
+        @Nullable
+        String keyEntryPassword;
+        @Nullable
+        String defaultKeyEntryAlias;
+        @Nullable
+        List<String> keyEntryAliases;
+        @Nonnull
+        String cipherName = CipherPool.DEFAULT_CIPHER;
+        @Nullable
+        SecureRandom secureRandom;
+
+        protected Builder() {
+        }
+
+        @Nullable
+        public String getKeyStoreFileName() {
+            return keyStoreFileName;
+        }
+
+        public void setKeyStoreFileName(@Nonnull String keyStoreFileName) {
+            this.keyStoreFileName = keyStoreFileName;
+        }
+
+        @Nullable
+        public String getKeyStorePassword() {
+            return keyStorePassword;
+        }
+
+        public void setKeyStorePassword(@Nonnull String keyStorePassword) {
+            this.keyStorePassword = keyStorePassword;
+        }
+
+        @Nullable
+        public String getKeyEntryPassword() {
+            return keyEntryPassword;
+        }
+
+        public void setKeyEntryPassword(@Nonnull String keyEntryPassword) {
+            this.keyEntryPassword = keyEntryPassword;
+        }
+
+        @Nullable
+        public String getDefaultKeyEntryAlias() {
+            return defaultKeyEntryAlias;
+        }
+
+        public void setDefaultKeyEntryAlias(@Nonnull String defaultKeyEntryAlias) {
+            this.defaultKeyEntryAlias = defaultKeyEntryAlias;
+        }
+
+        @Nullable
+        public List<String> getKeyEntryAliases() {
+            return keyEntryAliases;
+        }
+
+        public void setKeyEntryAliases(@Nonnull List<String> keyEntryAliases) {
+            this.keyEntryAliases = keyEntryAliases;
+        }
+
+        @Nonnull
+        public String getCipherName() {
+            return cipherName;
+        }
+
+        public void setCipherName(@Nonnull String cipherName) {
+            this.cipherName = cipherName;
+        }
+
+        @Nullable
+        public SecureRandom getSecureRandom() {
+            return secureRandom;
+        }
+
+        public void setSecureRandom(@Nonnull SecureRandom secureRandom) {
+            this.secureRandom = secureRandom;
+        }
+
+        @Nonnull
+        public KeyStoreSerializationKeyManager build() {
+            if (keyStoreFileName == null) {
+                throw new RecordCoreArgumentException("must specify key store file name");
+            }
+            if (keyStorePassword == null) {
+                keyStorePassword = "";
+            }
+            final KeyStore keyStore;
+            try {
+                keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
```

Comment:
Would it make sense to use:
```
 KeyStore ks = KeyStore.getInstance(new File(keyStoreFileName), keyStorePassword.toCharArray());
```

Then you don't need to open a `FileInputStream`, and call `load`

Reply:
Yes, that is simpler.